src/parquet/util/input.cc

```diff
@@ -134,7 +134,10 @@ void MemoryMapSource::Close() {
 }
 
 void MemoryMapSource::CloseFile() {
-  if (data_ != nullptr) { munmap(data_, size_); }
+  if (data_ != nullptr) {
+    munmap(data_, size_);
+    data_ = nullptr;
+  }
```

Comment:
Well, {{MemoryMapSource::Close}} is _always_ called at least twice. But the issue will only manifest in an environment where other threads are making concurrent {{mmap}} calls -- and even then, the manifestation is non-deterministic.

In general, the likelihood of manifestation should increase along with the number of threads making concurrent {{mmap}} and {{munmap}} calls. But in order for the issue to be triggered, a sequence of interleaved operations as illustrated in my commit message must occur (to include {{mmap}} having a tendency to reuse very recently unmapped segments, which I suspect could be platform-dependent).

All of that to say: the only reasonable testing approach I can think of here would be to call {{MemoryMapSource::Close}} multiple times for a single {{MemoryMapSource}} instance, and somehow confirm that {{munmap}} is only invoked once. The "cleanest" way to get such confirmation would be by intercepting {{munmap}} system calls...but I don't know how to do that off the top of my head.